keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types-to-staging.md

```diff
@@ -0,0 +1,296 @@
+
+#### Current kube-proxy
+
+* **Package**: [k8s.io/kubernetes/pkg/proxy/apis/kubeproxyconfig](https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/proxy/apis/kubeproxyconfig/types.go)
+* **GroupVersionKind**: `kubeproxy.config.k8s.io/v1alpha1.KubeProxyConfiguration`
+* **Supports** reading **config from file**, **without flag precedence**, **not tested**.
+* This API group embeds `ClientConnectionConfiguration`, which is **wrong**
+
+#### Current kube-scheduler
+
+* **Package**: [k8s.io/kubernetes/pkg/apis/componentconfig](https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/apis/componentconfig/types.go)
+* **GroupVersionKind**: `componentconfig/v1alpha1.KubeSchedulerConfiguration`
+* **Supports** reading **config from file**, **without flag precedence**, **not tested**
+* This API group embeds `ClientConnectionConfiguration` & `LeaderElectionConfiguration`, which is **wrong**
+
+#### Current kube-controller-manager
+
+* **Package**: [k8s.io/kubernetes/pkg/apis/componentconfig](https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/apis/componentconfig/types.go)
+* **GroupVersionKind**: `componentconfig/v1alpha1.KubeControllerManagerConfiguration`
+* **No support for config from file**
+* This API group embeds `ClientConnectionConfiguration` & `LeaderElectionConfiguration`, which is **wrong**
+
+#### Current kube-apiserver
+
+* **Doesn’t expose component configuration anywhere**
+* **No support for config from file**
+* The most similar thing to componentconfig for the API server is the `ServerRunOptions` struct in
+  [k8s.io/kubernetes/cmd/kube-apiserver/app/options/options.go](https://github.com/kubernetes/kubernetes/blob/release-1.11/cmd/kube-apiserver/app/options/options.go)
+
+#### Current cloud-controller-manager
+
+* **Package**: [k8s.io/kubernetes/pkg/apis/componentconfig](https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/apis/componentconfig/types.go)
+* **GroupVersionKind**: `componentconfig/v1alpha1.CloudControllerManagerConfiguration`
+* **No support for config from file**
+* This API group embeds `ClientConnectionConfiguration` & `LeaderElectionConfiguration`, which is **wrong**
+
+### Goals
+
+* Find a home for the ComponentConfig API types, hosted as a staging repo in the "core" repo that is kubernetes/kubernetes
+* Make ComponentConfig API types consumable from *projects outside of kube* and from different parts of kube itself
+    * Resolve dependencies from the external ComponentConfig API types so that everything can depend on them
+    * The only dependency of the ComponentConfig API types should be `k8s.io/apimachinery`
+* Split internal types from versioned types
+* Remove the monolithic `componentconfig/v1alpha1` API group
+* Enable the staging bot so that a `[https://github.com/kubernetes/](https://github.com/kubernetes/){component}`
+  (imported as `k8s.io/{component}`) repos are created (this also requires changes to the k8s.io Golang import meta data).
```

Comment:
The `go get` redirects are automatic now.

Reply:
fixed